dhcpv6/dhcpv6message.go

```diff
@@ -43,7 +43,7 @@ func GetTime() uint32 {
 }
 
 // NewSolicitWithCID creates a new SOLICIT message with CID.
-func NewSolicitWithCID(duid Duid, modifiers ...Modifier) (*Message, error) {
+func NewSolicitWithCID(duid Duid, hwaddr net.HardwareAddr, modifiers ...Modifier) (*Message, error) {
```

Comment:
How about removing `NewSolicitWithCID` and leaving only `NewSolicit` which takes a mac as parameter?

It doesn't make much sense to me to have a `NewSolicitWithCID` which takes a duid and a mac.

If we want to change the CID we can use the `WithClientID` modifier.

Reply:
Otherwise, the default duid will be DUID_LLT as specified in line 77.

Reply:
I am OK with that, I also think it makes more sense. CC @hugelgupf in case you may be impacted by this

Reply:
I am OK with that, I also think it makes more sense. CC @hugelgupf in case you may be impacted by this

Reply:
worth doing in a separate PR?

Reply:
updated